lemur/plugins/lemur_aws/acm.py

```diff
@@ -0,0 +1,178 @@
+    return True
+
+
+def get_id_from_arn(arn):
+    """
+    Extract the certificate name from an arn.
+
+    examples:
+    'arn:aws:acm:us-west-2:123456789012:certificate/1aa111a1-1a11-1111-11aa-a11aa111aa11' '1aa111a1-1a11-1111-11aa-a11aa111aa11'
+
+    :param arn: ACM TLS certificate arn
+    :return: id of the certificate as uploaded to AWS
+    """
+    return arn.split("/")[-1]
+
+
+@sts_client("acm")
+@retry(retry_on_exception=retry_throttled, wait_fixed=2000, stop_max_attempt_number=25)
+def upload_cert(name, body, private_key, cert_chain=None, **kwargs):
+    """
+    Upload a certificate to ACM AWS
+
+    :param body:
+    :param private_key:
+    :param cert_chain:
+    :param path:
+    :return:
+    """
+    assert isinstance(private_key, str)
+    client = kwargs.pop("client")
+
+    metrics.send("upload_cert", "counter", 1, metric_tags={"name": name})
+    try:
+        if cert_chain:
+            return client.import_certificate(
+                Certificate=str(body),
+                PrivateKey=str(private_key),
+                CertificateChain=str(cert_chain),
+            )
+        else:
+            return client.import_certificate(
+                Certificate=str(body),
+                PrivateKey=str(private_key),
+            )
+    except botocore.exceptions.ClientError as e:
+        if e.response["Error"]["Code"] != "EntityAlreadyExists":
+            raise e
+
+
+@sts_client("acm")
+@retry(retry_on_exception=retry_throttled, wait_fixed=2000, stop_max_attempt_number=25)
+def delete_cert(cert_arn, **kwargs):
+    """
+    Delete a certificate from ACM AWS
+
+    :param cert_arn:
+    :return:
+    """
+    client = kwargs.pop("client")
+    metrics.send("delete_cert", "counter", 1, metric_tags={"cert_arn": cert_arn})
+    try:
+        client.delete_certificate(CertificateArn=cert_arn)
+    except botocore.exceptions.ClientError as e:
+        if e.response["Error"]["Code"] != "NoSuchEntity":
+            raise e
+
+
+@sts_client("acm")
+def get_certificate(name, **kwargs):
+    """
+    Retrieves an acm SSL certificate.
+
+    :return:
+    """
+    return _get_certificate(name, **kwargs)
+
+
+@retry(retry_on_exception=retry_throttled, wait_fixed=2000, stop_max_attempt_number=25)
+def _get_certificate(arn, **kwargs):
+    metrics.send("get_certificate", "counter", 1, metric_tags={"arn": arn})
+    client = kwargs.pop("client")
+    try:
+        return client.get_certificate(CertificateArn=arn)
+    except client.exceptions.NoSuchEntityException:
+        capture_exception()
+        return None
+
+
+@sts_client("acm")
+def get_certificates(**kwargs):
+    """
+    Fetches one page of acm certificate objects for a given account.
+    :param kwargs:
+    :return:
+    """
+    return _get_certificates(**kwargs)
+
+
+@retry(retry_on_exception=retry_throttled, wait_fixed=2000, stop_max_attempt_number=25)
+def _get_certificates(**kwargs):
+    metrics.send("get_certificates", "counter", 1)
```

Comment:
here and elsewhere: Can we ensure this is namespaced under something ACM specific?

Reply:
Done!